File: src/extras-kguiaddons/libkcolormimedata.cpp
#include <KColorMimeData>
#include <QColor>
#include <QDrag>
#include <QMimeData>
#include <QObject>
#include <kcolormimedata.h>
#include "libkcolormimedata.h"
#include "libkcolormimedata.hxx"

void KColorMimeData_PopulateMimeData(QMimeData* param1, const QColor* param2) {
    KColorMimeData::populateMimeData(param1, *param2);
}

bool KColorMimeData_CanDecode(const QMimeData* param1) {
    return KColorMimeData::canDecode(param1);
}

QColor* KColorMimeData_FromMimeData(const QMimeData* param1) {
    return new QColor(KColorMimeData::fromMimeData(param1));
}

QDrag* KColorMimeData_CreateDrag(const QColor* param1, QObject* param2) {
    return KColorMimeData::createDrag(*param1, param2);
}

File: src_original/extras-kguiaddons/libkcolormimedata.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://api.kde.org/kcolormimedata.html
pub const kcolormimedata = struct {
    /// [Qt documentation](https://api.kde.org/kcolormimedata.html#populateMimeData)
    ///
    /// ``` param1: QtC.QMimeData, param2: QtC.QColor ```
    pub fn PopulateMimeData(param1: ?*anyopaque, param2: ?*anyopaque) void {
        qtc.KColorMimeData_PopulateMimeData(@ptrCast(param1), @ptrCast(param2));
    }

    /// [Qt documentation](https://api.kde.org/kcolormimedata.html#canDecode)
    ///
    /// ``` param1: QtC.QMimeData ```
    pub fn CanDecode(param1: ?*anyopaque) bool {
        return qtc.KColorMimeData_CanDecode(@ptrCast(param1));
    }

    /// [Qt documentation](https://api.kde.org/kcolormimedata.html#fromMimeData)
    ///
    /// ``` param1: QtC.QMimeData ```
    pub fn FromMimeData(param1: ?*anyopaque) QtC.QColor {
        return qtc.KColorMimeData_FromMimeData(@ptrCast(param1));
    }

    /// [Qt documentation](https://api.kde.org/kcolormimedata.html#createDrag)
    ///
    /// ``` param1: QtC.QColor, param2: QtC.QObject ```
    pub fn CreateDrag(param1: ?*anyopaque, param2: ?*anyopaque) QtC.QDrag {
        return qtc.KColorMimeData_CreateDrag(@ptrCast(param1), @ptrCast(param2));
    }
};
